python/pypio/pypio.py

```diff
@@ -15,9 +15,35 @@
 # limitations under the License.
 #
 
+from __future__ import absolute_import
+
+import sys
+
 from pypio.data import PEventStore
-from pypio.workflow import CleanupFunctions
+from pyspark.sql import SparkSession
+from pyspark.sql import utils
+
+
+def init():
+    global spark
+    spark = SparkSession.builder.getOrCreate()
+    global sc
+    sc = spark.sparkContext
+    global sqlContext
+    sqlContext = spark._wrapped
+    global p_event_store
+    p_event_store = PEventStore(spark._jsparkSession, sqlContext)
+    print("Initialized pypio")
+
+
+def find(app_name):
+    return p_event_store.find(app_name)
 
-p_event_store = PEventStore(spark._jsparkSession, sqlContext)
-cleanup_functions = CleanupFunctions(sqlContext)
 
+def save(model):
+    engine = sc._jvm.org.apache.predictionio.e2.engine.PythonEngine
+    engine.model().set(model._to_java())
+    main_args = utils.toJArray(sc._gateway, sc._gateway.jvm.String, sys.argv)
+    create_workflow = sc._jvm.org.apache.predictionio.workflow.CreateWorkflow
+    spark.stop()
+    create_workflow.main(main_args)
```

Comment:
This functions is wrapper of CreateWorkflow. It seems to can run even the evaluation by giving bootstrap arguments. Does it work?

Reply:
Python code to specify for the --main-py-file option is only launched when kicking off a training.

Reply:
Ah, top level commands don't accept illegal arguments so passing `sys.argv` directly to `CreateWorkflow.main()` is safe. I understand.